fs/dafilesrv/dafilesrv.cpp

```diff
@@ -642,6 +628,51 @@ int main(int argc, const char* argv[])
         exit(-1);
     }
 
+    {
+        Owned<IComponentLogFileCreator> lf = createComponentLogFileCreator(logDir.str(), "DAFILESRV");
+        lf->setCreateAliasFile(false);
+        lf->setMaxDetail(TopDetail);
+        lf->beginLogging();
+    }
+    write_pidfile(componentName.str());
+#else // _CONTAINERIZED
+    setupContainerizedLogMsgHandler();
+
+    dafileSrvInstance = config;
+
+    // k8s defaults, a bit arbitrary, but allow more concurrency by default than legacy
+    maxThreads = 400;
+    maxThreadsDelayMs = (60*1000);
+    maxAsyncCopy = 400;
+    parallelRequestLimit = 400;
+    throttleDelayMs = 1000;
+    throttleCPULimit = 85;
+    throttleQueueLimit = 1000;
+    parallelSlowRequestLimit = parallelRequestLimit;
+    throttleSlowDelayMs = throttleDelayMs;
+    throttleSlowCPULimit = throttleCPULimit;
+    throttleSlowQueueLimit = throttleQueueLimit;
+#endif
+
+    maxThreads = dafileSrvInstance->getPropInt("@maxThreads", maxThreads);
```

Comment:
Is there a chance in baremetal where we get here with dafileSrvInstance==nullptr?  Perhaps a misconfiguration with no Instance?

Reply:
as you perhaps if there's a slipped through misconfiguration.
I'll add a guard.